vyper/semantics/analysis/local.py

```diff
@@ -512,6 +513,10 @@ def visit_For(self, node):
 
         iter_var = None
         if isinstance(node.iter, vy_ast.Call):
+            if not isinstance(target_type, IntegerT):
+                raise TypeCheckFailure(
```

Comment:
i'd prefer to reuse the exception handling / error message from `validate_expected_type`

Reply:
we should also raise TypeMismatch (a regular exception) rather than a TypeCheckFailure (which is a panic)

hm yea the naming is confusing, we should really rename `TypeCheckFailure` to `TypeCheckPanic`

Reply:
Replaced  `TypeCheckFailure` by `TypeMismatch` and now use `validate_expected_type`